mocked-api/schema_types/blood_donation_center.py

```diff
@@ -3,39 +3,97 @@
 from typing import Optional
 from strawberry import relay
 from datetime import datetime
-from strawberry.types import Info
-from typing import Iterable
+
+# from strawberry.types import Info
+# from typing import Iterable
 from schema_types.blood_dotation_center_category import (
-    ENUM_BLOODDONATIONCENTER_CATEGORY,
+    BloodCenterCategory,
 )  # noqa: E501
+
 from schema_types.tag import Tag
 from schema_types.address import Address
 from schema_types.contacts import Contacts
 from schema_types.work_schedule_item import WorkScheduleItem
 
-# from mocked.mocked_blood_dotation_centers import mocked_data
-
 
 @strawberry.type
 class BloodDonationCenter(relay.Node):
+    attributes: strawberry.Private[object]
+
     id: relay.NodeID[int]
-    name: str
-    category: ENUM_BLOODDONATIONCENTER_CATEGORY
-    notes: Optional[str]
-    tags: typing.List[Tag]
-    address: Optional[Address]
-    contacts: Optional[Contacts]
-    workSchedule: Optional[typing.List[WorkScheduleItem]]
-    createdAt: Optional[datetime]
-    updatedAt: Optional[datetime]
-    publishedAt: Optional[datetime]
-
-    @classmethod
-    def resolve_nodes(
-        cls,
-        *,
-        info: Info,
-        node_ids: Iterable[str],
-        required: bool = False,
-    ):
-        return []  # mocked_data
+
+    # name
+    @strawberry.field
+    def name(self) -> str:
+        return str(self.attributes.get("name"))
+
+    # category
+    @strawberry.field
+    def category(self) -> BloodCenterCategory:
+        return BloodCenterCategory(self.attributes.get("category"))
+
+    # notes
+    @strawberry.field
+    def notes(self) -> Optional[str]:
+        return str(self.attributes.get("notes"))
+
+    # tags
+    @strawberry.field
+    def tags(self) -> Optional[typing.List[Tag]]:
+        tag_dicts = self.attributes.get("tags", {}).get("data", [])
+        for tag_dict in tag_dicts:
+            yield Tag(**tag_dict.get("attributes", {}))
+
+    # address
+    @strawberry.field
+    def address(self) -> Optional[Address]:
+        return Address(**self.attributes.get("address", {}))
+
+    # contacts
+    @strawberry.field
+    def contacts(self) -> Optional[Contacts]:
+        return Contacts(**self.attributes.get("contacts", {}))
+
+    # workSchedule
+    @strawberry.field
+    def workSchedule(self) -> Optional[typing.List[WorkScheduleItem]]:
+        for item in self.attributes.get("workSchedule"):
+            new_item = {
```

Comment:
If the intention of new item is to create a copy, you could say:
`new_item = deepcopy(item)`
And then do each check like:
```
if new_item.get("startTimeBeforeLunchBreak"):
    new_item["startTimeBeforeLunchBreak"] = new_item["startTimeBeforeLunchBreak"].strptime(TIME_FORMAT)
```
Also put "%H:%M:%S.%f" in a constant or even env variable if you have them. Time formattings are things that can change often, repeating them through the code would cause you to make many changes for a small thing.

Reply:
WorkSchedule data looks like this
```
"workSchedule": [
              {
                "weekDay": "Monday",
                "startTimeBeforeLunchBreak": "08:00:00.000",
                "endTimeBeforeLunchBreak": null,
                "startTimeAfterLunchBreak": null,
                "endTimeAfterLunchBreak": "14:00:00.000"
              },
              {
                "weekDay": "Tuesday",
                "startTimeBeforeLunchBreak": "08:00:00.000",
                "endTimeBeforeLunchBreak": null,
                "startTimeAfterLunchBreak": null,
                "endTimeAfterLunchBreak": "14:00:00.000"
              },
              {
                "weekDay": "Wednesday",
                "startTimeBeforeLunchBreak": "08:00:00.000",
                "endTimeBeforeLunchBreak": null,
                "startTimeAfterLunchBreak": null,
                "endTimeAfterLunchBreak": "14:00:00.000"
              },
              {
                "weekDay": "Thursday",
                "startTimeBeforeLunchBreak": "08:00:00.000",
                "endTimeBeforeLunchBreak": null,
                "startTimeAfterLunchBreak": null,
                "endTimeAfterLunchBreak": "14:00:00.000"
              },
              {
                "weekDay": "Friday",
                "startTimeBeforeLunchBreak": "08:00:00.000",
                "endTimeBeforeLunchBreak": null,
                "startTimeAfterLunchBreak": null,
                "endTimeAfterLunchBreak": "14:00:00.000"
              }
            ],
``` 
And idea is to transform time string values to actual datetime. Could you suggest more efficient way to do it?

Reply:
@IoanaTeoP  I made an optimized version for workSchedule. Please checl.

Reply:
Looks much better now 👍